From 81d7222b879d3e38c2a237ad98992d32f7f99aad Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 10 Dec 2023 17:49:50 -0800
Subject: [PATCH 1/2] Rename dirname local variable to dname

Ensures that it does not shadow the standard dirname function from libc.

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 edquota.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/edquota.c b/edquota.c
index a77106c..f326dc3 100644
--- a/edquota.c
+++ b/edquota.c
@@ -38,7 +38,7 @@ char *progname;
 static int flags, quotatype;
 static int fmt = -1;
 static char *protoname;
-static char *dirname;
+static char *dname;
 
 static void usage(void)
 {
@@ -138,7 +138,7 @@ static int parse_options(int argc, char **argv)
 				  exit(1);
 			  break;
 		  case 'f':
-			  dirname = optarg;
+			  dname = optarg;
 			  break;
 		  case 256:
 			  flags |= FL_NUMNAMES;
@@ -176,7 +176,7 @@ static void copy_prototype(int argc, char **argv, struct quota_handle **handles)
 	protoprivs = getprivs(protoid, handles, 0);
 	while (argc-- > 0) {
 		id = name2id(*argv, quotatype, !!(flags & FL_NUMNAMES), NULL);
-		curprivs = getprivs(id, handles, !dirname);
+		curprivs = getprivs(id, handles, !dname);
 		if (!curprivs)
 			die(1, _("Cannot get quota information for user %s\n"), *argv);
 		argv++;
@@ -223,7 +223,7 @@ int main(int argc, char **argv)
 	argv += ret;
 
 	init_kernel_interface();
-	handles = create_handle_list(dirname ? 1 : 0, dirname ? &dirname : NULL, quotatype, fmt,
+	handles = create_handle_list(dname ? 1 : 0, dname ? &dname : NULL, quotatype, fmt,
 			(flags & FL_NO_MIXED_PATHS) ? 0 : IOI_NFS_MIXED_PATHS,
 			(flags & FL_REMOTE) ? 0 : MS_LOCALONLY);
 	if (!handles[0]) {
@@ -296,7 +296,7 @@ int main(int argc, char **argv)
 	else {
 		for (; argc > 0; argc--, argv++) {
 			id = name2id(*argv, quotatype, !!(flags & FL_NUMNAMES), NULL);
-			curprivs = getprivs(id, handles, !dirname);
+			curprivs = getprivs(id, handles, !dname);
 			if (!curprivs)
 				die(1, _("Cannot get quota information for user %s.\n"), *argv);
 			if (flags & FL_EDIT_TIMES) {
-- 
2.43.0

